packages/template-retail-react-app/app/pages/checkout/partials/shipping-multi-address.jsx

```diff
@@ -0,0 +1,1245 @@
+                aria-live="polite"
+                aria-label={formatMessage(LOADING_LABEL)}
+            >
+                {formatMessage(LOADING_MESSAGE)}
+            </Box>
+        )
+    }
+
+    // Error state
+    if (productsError) {
+        return (
+            <Box
+                p={8}
+                textAlign="center"
+                color="red.500"
+                role="alert"
+                aria-live="assertive"
+                aria-label={formatMessage(ERROR_LABEL)}
+            >
+                {formatMessage(ERROR_MESSAGE)}
+            </Box>
+        )
+    }
+
+    const onAddNewAddress = () => {
+        // TODO: Implement add new address logic/modal
+        console.log('Add New Address clicked!')
+    }
+
+    // Keyboard navigation handler
+    const handleDropdownKeyDown = (e, addressKey, isMobile = false) => {
+        const dropdownKey = isMobile ? addressKey + '-mobile' : addressKey
+        const isOpen = openDropdown === dropdownKey
+        const totalOptions = addresses.length + 1 // +1 for "Add New Address"
+
+        if (!isOpen) {
+            if (e.key === 'Enter' || e.key === ' ' || e.key === 'ArrowDown') {
+                e.preventDefault()
+                setOpenDropdown(dropdownKey)
+                setFocusedIndex(0)
+            }
+        } else {
+            switch (e.key) {
+                case 'Escape':
+                    e.preventDefault()
+                    setOpenDropdown(null)
+                    setFocusedIndex(-1)
+                    break
+                case 'ArrowDown':
+                    e.preventDefault()
+                    setFocusedIndex((prev) => (prev + 1) % totalOptions)
+                    break
+                case 'ArrowUp':
+                    e.preventDefault()
+                    setFocusedIndex((prev) => (prev - 1 + totalOptions) % totalOptions)
+                    break
+                case 'Enter':
+                case ' ':
+                    e.preventDefault()
+                    if (focusedIndex === addresses.length) {
+                        // "Add New Address" option
+                        setOpenDropdown(null)
+                        setFocusedIndex(-1)
+                        onAddNewAddress()
+                    } else if (focusedIndex >= 0 && focusedIndex < addresses.length) {
+                        // Address option
+                        setSelectedAddresses((prev) => ({
+                            ...prev,
+                            [addressKey]: addresses[focusedIndex].addressId
+                        }))
+                        setOpenDropdown(null)
+                        setFocusedIndex(-1)
+                    }
+                    break
+                case 'Tab':
+                    setOpenDropdown(null)
+                    setFocusedIndex(-1)
+                    break
+            }
+        }
+    }
+
+    return (
+        <Box as="main" role="main" aria-label={formatMessage(MAIN_LABEL)}>
```

Comment:
It seems that you are re-implementing the select dropdown. We are using Chakra v2. Have you checked out Select component [here](https://v2.chakra-ui.com/docs/components/select)? I'd highly advise not reimplementing bc we have to think of a11y. Chakra v2 UI is a library that has a11y built in mind. We should use it instead of re-implementation the entire dropdown here.

If we have to build one, we should think about separating the logic of a DropDown Component into `components` folder and this component should be the one use the DropDown. This file is more than 1000+ of code and most of it is the implementation for Dropdown, and not much about multi-ship.

Reply:
Okay .. Will replace.